### PR TITLE
Nonuniform heatmaps are now possible with the GR backend.

### PR DESCRIPTION
**Pull request by lmh91**
Regarding the issue #1428.

### CODE REVIEW

**Comment by yha** [comment]
This breaks plotting heatmaps on non-linear scales. 
I think the support for directly specifying edges should be added by extending the `heatmap_edges` function (probably by a new function wrapping it) in `utils.jl`, so that it can be used by other backends as well. This can also avoid the code duplication in this PR, and deal with non-linear axes (as `heatmap_edges` already does).

**Comment by lmh91** [comment]
Good points!
I am still learning how Plots.jl works...
I hope I adjusted it in a proper way.

File: src/backends/gr.jl [hunk 1]
```diff
@@ -1317,23 +1314,32 @@ function gr_display(sp::Subplot{GRBackend}, w, h, viewport_canvas)
...
+                nx, ny = length(series[:x]), length(series[:y])
+                z_normalized = map(x -> GR.jlgr.normalize_color(x, zmin, zmax), z)
+                colors = Int32[round(Int32, 1000 + _i * 255) for _i in z_normalized]
+                xmin, xmax, ymin, ymax = xy_lims 
+                rmax = data_lims[4]
+                GR.setwindow(-rmax, rmax, -rmax, rmax)
+                if ymin > 0 
+                    @warn "'ymin[1] > 0' (rmin) is not yet supported."
+                end
+                @show series[:y][end]
```
Comment (daschw) [inline src/backends/gr.jl]: There's a left-over @show.

**Comment by oschulz** [comment]
Thanks for taking this on, Lukas!

**Pull request merged**